src/newgrf_roadtype.cpp

```diff
@@ -45,7 +46,7 @@
 			const Town *t = nullptr;
 			if (IsRoadDepotTile(this->tile)) {
 				t = Depot::GetByTile(this->tile)->town;
-			} else if (IsTileType(this->tile, MP_ROAD)) {
+			} else if (IsTileType(this->tile, MP_ROAD) || IsRoadStopTile(this->tile) || IsRoadTunnelBridge(this->tile)) {
```

Comment:
You need to check tile type before calling `IsRoadTunnelBridge(this->tile)`, or just add `IsRoadTunnelBridgeTile()`